src/main/kotlin/com/kotlindiscord/bot/Paginator.kt

```diff
@@ -0,0 +1,115 @@
+package com.kotlindiscord.bot
+
+import com.gitlab.kordlib.core.behavior.channel.MessageChannelBehavior
+import com.gitlab.kordlib.core.behavior.channel.createEmbed
+import com.gitlab.kordlib.core.behavior.edit
+import com.gitlab.kordlib.core.entity.Message
+import com.gitlab.kordlib.core.entity.ReactionEmoji
+import com.gitlab.kordlib.core.event.message.ReactionAddEvent
+import com.gitlab.kordlib.core.on
+import com.gitlab.kordlib.rest.builder.message.EmbedBuilder
+import kotlinx.coroutines.delay
+import mu.KotlinLogging
+
+private val FIRST_PAGE_EMOJI = ReactionEmoji.Unicode("\u23EE")
+private val LEFT_EMOJI = ReactionEmoji.Unicode("\u2B05")
+private val RIGHT_EMOJI = ReactionEmoji.Unicode("\u27A1")
+private val LAST_PAGE_EMOJI = ReactionEmoji.Unicode("\u23ED")
+private val DELETE_EMOJI = ReactionEmoji.Unicode("\u274C")
+
+private val EMOJIS = arrayOf(FIRST_PAGE_EMOJI, LEFT_EMOJI, RIGHT_EMOJI, LAST_PAGE_EMOJI, DELETE_EMOJI)
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * Interactive embed with multiple pages using emoji reactions as user inputs.
+ *
+ * NOTE : `.send()` needs to be called in order for the paginator to be displayed.
+ *
+ * @param kdBot Current instance of the bot.
+ * @param channel Channel to send the embed to.
+ * @param name Title of the embed.
+ * @param pages List of the embed pages.
+ * @param timeout Milliseconds before automatically deleting the embed. Set to a negative value to disable it.
+ * @param keepEmbed Keep the embed and only remove the reaction when the paginator is destroyed.
+ */
+class Paginator(
+    val kdBot: KDBot,
+    val channel: MessageChannelBehavior,
+    val name: String,
+    val pages: List<String>,
+    val timeout: Long = -1L,
+    val keepEmbed: Boolean = false
+) {
+    /** Message containing the embed. **/
+    var message: Message? = null
+    /** Current page of the paginator. **/
+    var currentPage: Int = 0
+    /** Should the paginator still process reaction events. **/
+    var doesProcessEvents: Boolean = true
+
+    /** Send the embed to the predefined channel parameter. **/
+    suspend fun send() {
+        val myFooter = EmbedBuilder.Footer()
+        myFooter.text = "Page 1/${pages.size}"
+        message = channel.createEmbed {
+            title = name
+            description = pages[0]
+            footer = myFooter
+        }
+        EMOJIS.forEach { message!!.addReaction(it) }
+        kdBot.bot.on<ReactionAddEvent> {
```

Comment:
Absolutely not, we need to use Live Entities for this. Adding handlers to the bot like this will result in a load of event handlers that never get cleaned up after we're done paginating the message.

https://gitlab.com/kordlib/kord/-/issues/34

Reply:
If I understand correctly, I just have to edit this line to
```kotlin
message.on<ReactionAddEvent> {
```
Right?

Reply:
I think so, but I haven't tried it myself. Might need to test it.